lib/new_relic/agent/log_event_aggregator.rb

```diff
@@ -50,9 +52,13 @@ def capacity
       end
 
       def record(formatted_message, severity)
-        @counter_lock.synchronize do
-          @seen += 1
-          @seen_by_severity[severity] += 1
+        return unless enabled?
+
+        if NewRelic::Agent.config[METRICS_ENABLED_KEY]
```

Comment:
Similar to how we memoize high security settings to a local bool in `register_for_done_configuring`, we could consider doing that as well for settings like this that are in the hot path (and other config checking below).

It's a bit ugly, but might be worth measuring whether it makes a difference or not in the microbenchmarks 🤔

Reply:
In a microbenchmarking test on hash access vs. instance variable memoization, it looks like there's a difference that falls within the margin of error for these options. Something similar is reflected when comparing these approaches in the log instrumentation performance tests. 

Since there isn't a difference, do you have a preference, @jasonrclark?

Reply:
Mild preference to just using the `config[]` if the data doesn't clearly support removing it. Thanks!